File: Pr6.java
import java.util.Scanner;
public class Pr6 {

    public static String reverseString(String s) {
        int length = s.length();
        StringBuilder reversed = new StringBuilder(length);

        for (int i = length - 1; i >= 0; i--) {
            reversed.append(s.charAt(i));
        }

        return reversed.toString();
    }

    public static void main(String[] args) {
        String s;
        Scanner inp= new Scanner(System.in);
        System.out.print("Enter a string: ");
        s = inp.nextLine();
        
        String revStr = reverseString(s);

        System.out.println("Original string: " + s);
        System.out.println("Reversed string: " + revStr);
    }
}

File: Assignment_1_cpp/Pr6.cpp
#include <iostream>
using namespace std;

int main()
{
	string s;
	char swap;
	
	cout<<"Enter a string: ";
	getline(cin,s);
	cout<<endl<<"Original string is: "<<s<<endl;
	
	int l;
	for(l=0;s[l];l++);
	
	for(int i=0;i<l/2;i++)
	{
		swap = s[i];
		s[i] = s[l-1-i];
		s[l-1-i] = swap;
	}
	cout<<"Reversed string is: "<<s;
}